www/front_src/src/Navigation/Sidebar/Menu/index.tsx

```diff
@@ -0,0 +1,256 @@
+  propsSelectedNavigationItems,
+} from '../sideBarAtoms';
+import { closedDrawerWidth, openedDrawerWidth } from '../index';
+
+import CollapsableItems, { collapseBorderWidth } from './CollapsableItems';
+import MenuItems from './MenuItems';
+import icons from './icons';
+
+interface Props {
+  isDrawerOpen: boolean;
+  navigationData?: Array<Page>;
+}
+
+const useStyles = makeStyles((theme) => ({
+  icon: {
+    color: theme.palette.text.primary,
+    fontSize: 26,
+  },
+  list: {
+    '&.MuiList-root': {
+      padding: theme.spacing(0, 0, 0, 0),
+    },
+  },
+}));
+
+const NavigationMenu = ({
+  isDrawerOpen,
+  navigationData,
+}: Props): JSX.Element => {
+  const classes = useStyles();
+  const navigate = useNavigate();
+  const { pathname, search } = useLocation();
+
+  const [hoveredIndex, setHoveredIndex] = useState<number | null>(null);
+  const [currentTop, setCurrentTop] = useState<number>();
+  const [collapseScrollMaxHeight, setCollapseScrollMaxHeight] = useState<
+    number | undefined
+  >(undefined);
+  const [collapseScrollMaxWidth, setCollapseScrollMaxWidth] = useState<
+    number | undefined
+  >(undefined);
+  const [selectedNavigationItems, setSelectedNavigationItems] = useAtom(
+    selectedNavigationItemsAtom,
+  );
+  const [itemsHoveredByDefault, setItemsHoveredByDefault] = useAtom(
+    itemsHoveredByDefaultAtom,
+  );
+  const levelName = 'level_0_Navigated';
+  const currentWidth = isDrawerOpen ? openedDrawerWidth / 8 : closedDrawerWidth;
+
+  const props = {
+    collapseScrollMaxHeight,
+    collapseScrollMaxWidth,
+    currentTop,
+    currentWidth,
+    hoveredIndex,
+    isDrawerOpen,
+    level: 1,
+    pathname,
+    search,
+    setCollapseScrollMaxHeight,
+    setCollapseScrollMaxWidth,
+  };
+
+  const hoverItem = (
+    e: React.MouseEvent<HTMLElement>,
+    index: number | null,
+    item: Page,
+  ): void => {
+    const rect = e.currentTarget.getBoundingClientRect();
+    const { top } = rect;
+    setCurrentTop(top - collapseBorderWidth);
+    setHoveredIndex(index);
+    setSelectedNavigationItems({
+      ...selectedNavigationItems,
+      level_0: { index, label: item.label, url: item?.url },
+    });
+  };
+
+  const handleLeave = (): void => {
+    setHoveredIndex(null);
+  };
+
+  const getUrlFromEntry = (entryProps: Page): string | null | undefined => {
+    const page = !isNil(entryProps?.page) ? entryProps.page : '';
+    const options = !isNil(entryProps?.options) ? entryProps.options : '';
+
+    const urlOptions = `${page}${options}`;
+    const url = entryProps.is_react
+      ? entryProps.url
+      : `/main.php?p=${urlOptions}`;
+
+    return url;
+  };
+
+  const addSelectedNavigationItems = (
+    navigationItems: Record<string, propsSelectedNavigationItems>,
+    level: number,
+  ): Record<string, propsSelectedNavigationItems> => {
+    const updatedNavigationItems = clone(navigationItems);
+
+    keys(updatedNavigationItems).forEach((i: string) => {
```

Comment:
You can use the `map` function from Ramda https://ramdajs.com/docs/#map
It loops throughout object